### ISSUE TITLE
Folding C/C++ `else if` is unnatural

### ISSUE THREAD

**Issue by ecnerwala** [nvim-treesitter/nvim-treesitter#1567]
**Describe the bug**

Folding an else-if clause is pretty unnatural: an if/else-if/else-if/else-if/else block is parsed as a nested sequence of if/else(if-else(...)) blocks, which makes sense for the grammar but not for humans reading the code. This creates the observed behavior that fold is created for each "suffix", except not for the trailing else block, which feels both unnatural and a little inconsistent.

We should probably take `foldmethod=syntax`'s approach and fold the entire block as one fold.

**To Reproduce**

```
int main() {
	if (1) {
		// stuff
	} else if (1) {
		// stuff
	} else if (1) {
		// stuff
	} else if (1) {
		// stuff
	} else if (1) {
		// stuff
	} else {
		// stuff
	}
}
```

**Comment by theHamsta** [comment]
I agree. I think the only way how this could be handled would be to define folds for `ifs` that whose parent is not a `else` block. One would have to hard code folds for a fixed numbers of else blocks.  Another solution would be to change the grammar of the parser to have the notion of a `else if`.
